src/Builder.php

```diff
@@ -73,6 +73,15 @@ public function withHeader(string $name, mixed $value): Builder;
      */
     public function withClaim(string $name, mixed $value): Builder;
 
+    /**
+     * Configures multiple claim items
+     *
+     * @param non-empty-array $claims
+     *
+     * @throws RegisteredClaimGiven When trying to set a registered claim.
+     */
+    public function withClaim(array $claims): Builder;
```

Comment:
Adding methods to an interface is a major breakage, and it must **REALLY** be worth doing. For utility methods, BC breaks are simply not worth pursuing.

This can be done in userland with an `array_reduce()` or similar, if really needed.

Reply:
Can you please give me an example, how one can setup multiple claims with a for loop in case you have this code:
`
    $tokenBuilder   = new Builder(new JoseEncoder(), ChainedFormatter::default());
    $signingKey     = InMemory::file(__DIR__ . $config['api_key']);
    $id                     = getTokenId();
    $now                = new DateTimeImmutable();
    $token             = $tokenBuilder
        ->issuedBy($config['issued_by'])
        ->permittedFor($config['permitted_for'])
        ->identifiedBy($id)
        ->issuedAt($now)
        ->canOnlyBeUsedAfter($now)
        ->expiresAt($now->modify('+1 hour'))
        ->withHeader('module', $module)
        ->withClaims($claims)
        ->getToken(new Sha256(), $signingKey);
`
You can not add them to $tokenBuilder prior or after generating the token. So want to understand how can I implement it without adding withClaims()?

I'm using Signer\Ecdsa\Sha256 if that is of any difference.

Reply:
pseudo-code, but it's this:


```php
foreach ($claims as $name => $claim) {
    $builder = $builder->withClaim($name, $claim);
}
```

Reply:
thank you!

Reply:
@hackman you can also use our high-level API to simplify your code: https://lcobucci-jwt.readthedocs.io/en/stable/quick-start/#issuing-tokens

```php
$token = (new JwtFacade())->issue(
    new Sha256(),
    InMemory::file(__DIR__ . $config['api_key']),
    static function (Builder $builder, DateTimeImmutable $issuedAt) use ($config): Builder {
        foreach (['claim1' => 'value1'] as $claim => $value) {
            $builder = $builder->withClaim($claim, $value);
        }

        return $builder
            ->issuedBy($config['issued_by'])
            ->permittedFor($config['permitted_for'])
            ->identifiedBy(getTokenId())
            ->expiresAt($issuedAt->modify('+1 hour')) // default is 5 minutes
            ->withHeader('module', $module);
    },
);
```